### PR TITLE
Moved RavenDistinctProject class  from mig to calcite

### PR DESCRIPTION
**Pull request by saurabhgupta-datametica**
Moved RavenDistinct project from mig to calcite and andded logic to make new rel for distict aggregate type of rel

### CODE REVIEW

**Comment by vjain061** [comment]
LGTM

**Comment by vjain061** [comment]
Please check these failure.

**Review by vjain061 — APPROVED** [review]
L1

File: core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.function.Supplier;
+
+/**
+ * RavenDistinctProject is an extension of Project which represent distinct by a boolean value.
+ */
+
+
+public class RavenDistinctProject extends Project {
+
+  boolean distinct;
```
Comment (abbas-gadhia) [inline core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java:45]: Hi @praveendatametica 
I dont think this is the right way to model a distinct. Please discuss.

Comment (praveendatametica) [reply core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java:45]: Sure , will discuss the same internally

Comment (saurabhgupta-datametica) [reply core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java:45]: We are not adding any logic here just moving RavenDistinctProject from mig to calcite.
Earlier we used to have distinct flag in project but in new update this flag is removed , so to figure out distinct rel node we have moved this class.

I know this is not the best approach , but for the time being can we proceed with merge , and parallelly  discuss the right way to figure out distinct in calcite

Comment (abbas-gadhia) [reply core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java:45]: I dont think there was ever a flag in the Project for the distinct boolean. The fields used in the project were used to create a LogicalAggregate whose group by fields were the fields of the Project. You can see this code in RelBuilder.

We already have an approach to identify a distinct i.e look to see if the project fields are the same as the aggregate fields. We should proceed with it.

Comment (kk19ag) [reply core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java:45]: Hi Abbas, so converting the LogicalAgg -> LogicalProject thing back to a Distinct keyword will be done(already exists or we can do something to make it so) OR are you suggesting we should  generate `select id, name from t1 group by id, name ` to achieve DISTINCT behaviour ? 

Also, there was a isDistinct() behaviour in LogicalProject before the big update in OUR calcite version which this flag was overriding. This RavenDistinctProject class was added in that pre update phase. Now the API got removed but this class still exists.
```
@SuppressWarnings("deprecation")
  public boolean isDistinct() {
    final RelMetadataQuery mq = RelMetadataQuery.instance();
    return Boolean.TRUE.equals(mq.areRowsUnique(this));
  }
```

**Comment by kk19ag** [comment]
merging and following up.

**Pull request merged**